ndcube/utils/wcs.py

```diff
@@ -138,77 +140,118 @@ def _wcs_slicer(wcs, missing_axis, item):
 
     missing_axis: `list` of `bool`
         Altered missing axis list.  Note the ordering has been reversed to reflect the data
-        (numpy) axis ordering convention.
+        (numpy) axis ordering convention. 
+        
+    dropped_coords:
+        Coordinates which have been dropped in the slicing process is collected in a tuple called
+        `dropped_coords`. 
 
     """
+    # Check ordering. 
+    if numpy_order is True:
+        item = item[::-1]
     # normal slice.
     item_checked = []
     if isinstance(item, slice):
         index = 0
-        # creating a new tuple of slice where if the axis is dead i.e missing
-        # then slice(0,1) added else slice(None, None, None) is appended and
+        # Creating a new tuple of slice where if the axis is dead, i.e. missing,
+        # then slice(0,1) added, else slice(None, None, None) is appended; and,
         # if the check of missing_axis gives that this is the index where it
         # needs to be appended then it gets appended there.
-        for _bool in missing_axis:
+        for i, _bool in enumerate(missing_axis):
             if not _bool:
-                if index is not 1:
+                if index is not 1:  
                     item_checked.append(item)
                     index += 1
                 else:
                     item_checked.append(slice(None, None, None))
             else:
                 item_checked.append(slice(0, 1))
-        new_wcs = wcs.slice((item_checked))
+        item_ = (item_checked)
     # item is int then slicing axis.
     elif isinstance(item, int) or isinstance(item, np.int64):
-        # using index to keep track of whether the int(which is converted to
+        # Using index to keep track of whether the int(which is converted to
         # slice(int_value, int_value+1)) is already added or not. It checks
-        # the dead axis i.e missing_axis to check if it is dead than slice(0,1)
+        # the dead axis i.e. missing_axis to check if it is dead than slice(0,1)
         # is appended in it. if the index value has reached 1 then the
         # slice(None, None, None) is added.
         index = 0
         for i, _bool in enumerate(missing_axis):
             if not _bool:
                 if index is not 1:
-                    item_checked.append(slice(item, item+1))
-                    missing_axis[i] = True
+                    item_checked.append(slice(item, item + 1))
                     index += 1
                 else:
                     item_checked.append(slice(None, None, None))
-            else:
+            elif _bool: 
```

Comment:
This line can simply be ```else:``` as there are only two options, ```bool_``` and ```not bool_```.  We already checked ```if not bool_```, so no need to evaluate it again here.

This makes very little practical difference in this case.  It's just a slightly better style.